features/370-auth-ui/design.md

```diff
@@ -0,0 +1,331 @@
+# Auth UI
+
+## Background
+
+Auth UI is a web application in Auth Gear. It aims to become a OpenID Provider.
+
+## Auth Gear acting as OP
+
+After the introduction of Auth UI, Auth Gear acts as an OpenID Provider (OP). API Client is now treated as Relying Party (RP).
+
+Authorization in the context of OpenID Connect is the authorized scopes the user has granted to the RP. Only the value `openid` is officially defined. In the context of Skygear Auth, possible values of scope include `email` and `phone_number`. They correspond to the value of the login ID. If the `email` scope is granted and the identity the user is using does have the `email` claim, then the value is included in the UserInfo Response.
+
+Auth Gear does not support Authorization at the moment. To be backward compatible, existing access tokens can access Data without any restriction. That means the user can change password, list sessions of **ALL** API clients, etc, via Auth Gear. The new access tokens issued by Auth Gear are intended to be consumed in the same fashion so the new and the old behave the same.
+
+In the future Auth Gear can act as an OP for another Skygear App. At that moment Authorization must be introduced. Existing API clients are considered as first party and have the largest scopes. The RP Skygear App must register itself as a new kind of API client with the OP Skygear App. This is analogous to using access token issued by Google cannot access Google API to change Google account password unless the user authorize the corresponding scope.
+
+## Domain
+
+Auth Gear is served in its own domain. It also also be served in a designated custom domain.
+
+### Default domain
+
+If the app is using default domain, then Auth Gear is served at `https://accounts.<appname>.skygearapp.com`. Note that this domain format may conflict with deployment version. Reserved label has higher precedence than deployment version. The choice of `accounts` is borrowed from some other internet services.
+
+### Custom domain
+
+If the app is using custom domain, then Auth Gear is served at the developer's wish.
```

Comment:
No sure how custom domain looks like after this change? Originally we only have one domain for all gears and Skygear App. 

After auth ui is supported, developers can configure gears domain and skygear app domain separately?

Reply:
Yes. I should update the spec of custom domain to reflect this.